### PR TITLE
python3: Bump version 3.11 to 3.11.9

### PR DESCRIPTION
**Pull request by qosmio**
Also bumps setuptools to version 24.0

Run tested: aarch64, Dynalink DL-WRX36, Master Branch

Maintainer: Jeffery To <jeffery.to@gmail.com>

### CODE REVIEW

**Comment by Ra2-IFV** [comment]
You need to update patches. Run `make package/path/to/package/refresh` and check this [doc](https://openwrt.org/docs/guide-developer/toolchain/use-patches-with-buildsystem) for more details.

**Comment by qosmio** [comment]
Oops, thought I had. Thanks for catching this. Just pushed the refreshed version.

**Comment by Ra2-IFV** [comment]
Looks great!

**Review by commodo — APPROVED** [review]

**Comment by commodo** [comment]
Will let @jefferyto handle this :)

I just added a quick review

**Comment by neheb** [comment]
```
2024-06-30T02:09:56.7237524Z /builder/build_dir/hostpkg/Python-3.11.9/Modules/_ssl.c: In function 'x509_object_dup':
2024-06-30T02:09:56.7251473Z /builder/build_dir/hostpkg/Python-3.11.9/Modules/_ssl.c:4554:18: error: implicit declaration of function 'X509_OBJECT_set1_X509'; did you mean 'X509_OBJECT_get0_X509'? [-Werror=implicit-function-declaration]
2024-06-30T02:09:56.7253411Z  4554 |             ok = X509_OBJECT_set1_X509(ret, X509_OBJECT_get0_X509(obj));
2024-06-30T02:09:56.7254222Z       |                  ^~~~~~~~~~~~~~~~~~~~~
2024-06-30T02:09:56.7255226Z       |                  X509_OBJECT_get0_X509
2024-06-30T02:09:56.7337478Z /builder/build_dir/hostpkg/Python-3.11.9/Modules/_ssl.c:4558:18: error: implicit declaration of function 'X509_OBJECT_set1_X509_CRL'; did you mean 'X509_OBJECT_get0_X509_CRL'? [-Werror=implicit-function-declaration]
2024-06-30T02:09:56.7339489Z  4558 |             ok = X509_OBJECT_set1_X509_CRL(
2024-06-30T02:09:56.7340166Z       |                  ^~~~~~~~~~~~~~~~~~~~~~~~~
2024-06-30T02:09:56.7340828Z       |                  X509_OBJECT_get0_X509_CRL
2024-06-30T02:09:56.7419780Z /builder/build_dir/hostpkg/Python-3.11.9/Modules/_ssl.c: In function 'X509_STORE_get1_objects':
2024-06-30T02:09:56.7434658Z /builder/build_dir/hostpkg/Python-3.11.9/Modules/_ssl.c:4578:10: error: implicit declaration of function 'X509_STORE_lock'; did you mean 'X509_STORE_free'? [-Werror=implicit-function-declaration]
2024-06-30T02:09:56.7436494Z  4578 |     if (!X509_STORE_lock(store)) {
2024-06-30T02:09:56.7437079Z       |          ^~~~~~~~~~~~~~~
2024-06-30T02:09:56.7437591Z       |          X509_STORE_free
2024-06-30T02:09:56.7630625Z /builder/build_dir/hostpkg/Python-3.11.9/Modules/_ssl.c:4581:11: error: implicit declaration of function 'sk_X509_OBJECT_deep_copy'; did you mean 'sk_X509_OBJECT_pop'? [-Werror=implicit-function-declaration]
2024-06-30T02:09:56.7632708Z  4581 |     ret = sk_X509_OBJECT_deep_copy(X509_STORE_get0_objects(store),
2024-06-30T02:09:56.7633558Z       |           ^~~~~~~~~~~~~~~~~~~~~~~~
2024-06-30T02:09:56.7634151Z       |           sk_X509_OBJECT_pop
2024-06-30T02:09:56.7648130Z /builder/build_dir/hostpkg/Python-3.11.9/Modules/_ssl.c:4581:9: warning: assignment to 'struct stack_st_X509_OBJECT *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
2024-06-30T02:09:56.7650168Z  4581 |     ret = sk_X509_OBJECT_deep_copy(X509_STORE_get0_objects(store),
2024-06-30T02:09:56.7650959Z       |         ^
2024-06-30T02:09:56.7772643Z /builder/build_dir/hostpkg/Python-3.11.9/Modules/_ssl.c:4583:5: error: implicit declaration of function 'X509_STORE_unlock'; did you mean 'X509_STORE_up_ref'? [-Werror=implicit-function-declaration]
2024-06-30T02:09:56.7774505Z  4583 |     X509_STORE_unlock(store);
2024-06-30T02:09:56.7775061Z       |     ^~~~~~~~~~~~~~~~~
2024-06-30T02:09:56.7775545Z       |     X509_STORE_up_ref
```

tools/libressl needs updating I think.

**Comment by neheb** [comment]
aaand looks like that wouldn't fix it.

**Comment by qosmio** [comment]
@neheb are you building OpenSSL with `CONFIG_OPENSSL_NO_DEPRECATED=y`?

**Comment by neheb** [comment]
no this is the host build. check CI logs.

**Comment by qosmio** [comment]
Please try with latest push. Added additional patch for host build.

https://github.com/openwrt/packages/compare/master...qosmio:packages:python3-3.11.9#diff-6b743dc2741d04bf68b3a006077677f8bd8330febc07acb6ead9159761051ba8

**Comment by neheb** [comment]
It's unfortunate really. Upstream killed compatibility with LibreSSL. Would be great to kill tools/libressl. Unfortunately openssl has a horrible build system. Although I've heard they're looking to switching to meson.

File: lang/python/python3/patches/027-fix-host-build-libressl.patch [hunk 1]
```diff
@@ -0,0 +1,35 @@
+--- a/Modules/_ssl.c
++++ b/Modules/_ssl.c
```
Comment (BKPepe) [inline lang/python/python3/patches/027-fix-host-build-libressl.patch:2]: Commit and neither commit description says anything about this patch. Is it part of upstream repository?

**Review by BKPepe — CHANGES_REQUESTED** [review]

**Comment by BKPepe** [comment]
Backported to OpenWrt 24.10 - 9684653566522b5616629d39d797cb0756217502

File: lang/python/python3/patches/027-fix-host-build-libressl.patch (hunk 1 above)
Comment (BKPepe) [reply lang/python/python3/patches/027-fix-host-build-libressl.patch:2]: ping @qosmio

**Pull request merged**